src/iop/toneequal.c

```diff
@@ -1017,7 +1017,10 @@ void toneeq_process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece,
   if(self->dev->gui_attached && (piece->pipe->type & DT_DEV_PIXELPIPE_FULL) == DT_DEV_PIXELPIPE_FULL)
   {
     if(g->mask_display)
+    {
       display_luminance_mask(in, luminance, out, roi_in, roi_out, ch);
+      piece->pipe->mask_display = DT_DEV_PIXELPIPE_DISPLAY_PASSTHRU;
+    }
     else
       apply_toneequalizer(in, luminance, out, roi_in, roi_out, ch, d);
```

Comment:
Don't you need to reset `piece->pipe->mask_display` in the `else` branch ?

Reply:
Not necessary, it gets reset by the pixelpipe code.  The mask display properly disappears when toneequal is turned off or loses focus, or the mask button is clicked again.  Neither colorzones nor retouch explicitly reset mask_display.